lib/libzutil/zutil_import.c

```diff
@@ -515,6 +515,20 @@ get_configs(libpc_handle_t *hdl, pool_list_t *pl, boolean_t active_ok,
 	for (pe = pl->pools; pe != NULL; pe = pe->pe_next) {
 		uint64_t id, max_txg = 0;
 
+		/*
+		 * These are to silence complaints from Clang's static
+		 * analyzer.  It claims that if we read an invalid
+		 * configuration that contains no top level vdevs, the operand
+		 * of a comparison is a garbage value. This should never
+		 * happen, but if it somehow does, breaking get_confgs() with
+		 * an assertion would be undesireable, since it would break the
+		 * userland tools for all valid pools, so instead of an
+		 * assertion, we just initialize these and document the reason
+		 * why.
+		 */
+		holes = 0;
+		hostid = 0;
```

Comment:
Pulling the declaration and initialization of `holes` and `hostid` in to the for loop seems like it would also make Clang happy and perhaps be more readable. There might be a few other local variables we could reduce the scope of as well.  i.e.

```diff
        for (pe = pl->pools; pe != NULL; pe = pe->pe_next) {
                uint64_t id, max_txg = 0;
+               uint64_t hostid = 0;
+               uint_t holes = 0;
```

Reply:
That would cause Clang to complain again, since Clang is concerned about the case that the loop does not iterate.

Reply:
I'm not sure I follow.  In the case where the top-level vdev loop isn't iterated they would still be initialized at the top of the outer loop.

Reply:
That is where they are now.

Reply:
Actually, I see what you mean now. I will move declaration into the loop.

Reply:
Done.